### PR TITLE
Kayrus optional parameters

### PR DESCRIPTION
**Pull request by megan07**
Addressing #230, #205, and #190 where GITHUB_TOKEN should not be required for all data sources.
Before fix:
```
        Error: GET https://api.github.com/users/hashibot: 401 Bad credentials []

          on /var/folders/nj/n5hr70hd2dx45qb__6wl10nc0000gp/T/tf-test353694711/main.tf line 6:
          (source code not available)


FAIL
```
After fix:
```
== RUN   TestProvider_anonymous
--- PASS: TestProvider_anonymous (1.04s)
PASS
```

### CODE REVIEW

File: github/provider.go [hunk 1]
```diff
@@ -39,6 +39,12 @@ func Provider() terraform.ResourceProvider {
 				Default:     false,
 				Description: descriptions["individual"],
 			},
+			"anonymous": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
```
Comment (paddycarver) [inline github/provider.go]: This will default to falsey anyways, so we can avoid having this Default here.

File: github/provider.go [hunk 2]
```diff
@@ -92,6 +99,8 @@ func init() {
 			"without verifying the TLS certificate.",
 
 		"individual": "Whether to run outside an organization.",
+
+		"anonymous": "Whether to authenticate without a token.",
```
Comment (paddycarver) [inline github/provider.go]: My problem with "whether or not to do X" descriptions of booleans is the cognitive hurdle it takes to determine whether `true` or `false` leads to X actually happening. `Set to true to authenticate without a token` avoids this shortcoming, as an example.

File: github/provider_test.go [hunk 3]
```diff
@@ -90,6 +102,59 @@ func TestProvider_individual(t *testing.T) {
 				Config:      individualProviderConfig + testAccGithubMembershipConfig(username),
 				ExpectError: regexp.MustCompile("This resource requires GitHub organization to be set on the provider."),
 			},
+			{
+				Config:      conflictingProviderConfig + testAccCheckGithubUserDataSourceConfig(username),
+				ExpectError: regexp.MustCompile("If `individual` is true, `organization` cannot be set."),
```
Comment (paddycarver) [inline github/provider_test.go]: Does this test pass? `conflictingProviderConfig`, from what I can see, only sets `individual` to `true`, it doesn't set an organization.

File: github/provider_test.go [hunk 4]
```diff
@@ -71,6 +71,18 @@ func TestProvider_individual(t *testing.T) {
...
 }
 `
+
+	conflictingProviderConfig := `provider "github" {
+	individual = true
+}
+`
+
+	neitherProviderConfig := `provider "github" {
+	organization = ""
```
Comment (paddycarver) [inline github/provider_test.go]: I think the more common user scenario, and the one we should probably be testing for, is that they didn't specify organization at all, rather than setting it to an empty string.

Comment (megan07) [reply github/provider_test.go]: @paddycarver I agree, however, the rest of the tests need GITHUB_ORGANIZATION defined, so, when running the suite it will already be there, this will override that.  Is there a better way to do this?

**Comment by kayrus** [comment]
@megan07 when are you going to make the release with this feature?
